crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -104,11 +104,17 @@ pub fn compile_circuit_and_witness<P: AsRef<Path>>(
 
     let solver_res = backend.solve(&mut solved_witness, compiled_program.circuit.gates.clone());
 
-    if let Err(opcode) = solver_res {
-        return Err(CliError::Generic(format!(
-            "backend does not currently support the {} opcode. ACVM does not currently fall back to arithmetic gates.",
-            opcode
-        )));
+    if let Err(gate_error) = solver_res {
+        match gate_error {
```

Comment:
May be a bit simpler if we matched on solver_res directly. The Ok branch could just return what is currently on line 119

Reply:
Done